### PR TITLE
Cleanup testable framework

### PR DESCRIPTION
**Pull request by hardikmaheshwari**
#### What type of PR is this?
Cleanup

#### What does this PR do / why is it needed ?
Cleanup testable framework

Depends on : https://github.com/finos/legend-pure/pull/642

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

**Review by MauricioUyaguari — COMMENTED** [review]
Looks good to me. One small thing, I think we are missing the `ProtocolSubTypeInfo` for TestExecuted so `_type` is added when we seralize to JSON.

File: legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/test/result/TestExecuted.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package org.finos.legend.engine.protocol.pure.v1.model.test.result;
+
+import org.finos.legend.engine.protocol.pure.v1.model.test.assertion.status.AssertionStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class TestExecuted extends TestResult
+{
+    public List<AssertionStatus> assertStatuses = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/test/result/TestExecuted.java:24]: @hardikmaheshwari answered my own question when looking at the code :)

[the remark this answers is not in the document]
Comment (hardikmaheshwari) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/test/result/TestExecuted.java:27]: @MauricioUyaguari Had a quick check checkpoint with @pierredebelen yesterday. He suggested we don't need inheritance for TestPass and TestFail. Can you please confirm if we this changes are possible to incorporate in Studio

[the remark this answers is not in the document]
Comment (MauricioUyaguari) [reply legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/test/result/TestExecuted.java:27]: looks good.

**Review by MauricioUyaguari — APPROVED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

**Review by gs-ssh16 — APPROVED** [review]

**Pull request merged**